v2_explainer.md

```diff
@@ -0,0 +1,76 @@
+# Picture-in-Picture V2 Explained
+
+## Background: Picture-in-Picture V1
+
+Picture-in-Picture launched for HTMLVideoElement early in 2019. The API was
+designed to be extensible, like the Fullscreen API but at the moment is only
+defined for video elements. The implementation takes the video layer and moves
+it into a window that follows display rules specific to Picture-in-Picture
+(always on top, fixed aspect ratio, etc.).
+
+## What's Missing from V1?
+
+Various partners expressed frustration around the limitations of the API. The
+feature requests fell into two categories: customising the look & feel and
+customising the user experience. The former can technically be done using canvas
+elements but the latter requires a different model that would require user
+interaction and therefore no longer be compatible with Android, iOS and macOS
+APIs.
+
+## Goals
+
+  1. **Allow sites to Picture-in-Picture arbitary content**
+  2. **A site should be able to choose whether that content is interactive.** If
+the platform (e.g. Android) does not support interactive content then the site
+should be able to easily fallback to non-interactive content
+
+## Non-Goals
+
+The Picture-in-Picture window is not a popup. Therefore, the content should not
+live beyond the lifetime of the page.
+
+## Sample Usage
+
+Here's how a page might request Picture-in-Picture for it's media controls and
+then move the video element to the new window. When the window is closed it will
+then move the video element back to the page.
+
+```html
+<video id=video>
```

Comment:
I naively thoughts most people would simply request PiP for the entire video player element and have CSS media queries to customize it.

```html
<div id="video-player">
  <div id="video-controls">...</div>
  <video id="video">...</video>
</div>
```
```js
const player = document.querySelector('#video-player');
const pipWindow = await player.requestPictureInPicture({
  aspectRatio: 1,
  width: 200,
  height: 200,
  interactive: true
});
```

Reply:
The `requestPictureInPicture` copies the DOM element to the new page so it would break video playback. Having an exception to move elements in a copy algorithm might be a bit confusing. We could move the DOM element however we would then not have an element to set for `pictureInPictureElement`.

Since the new window is also a new document it would have its own set of stylesheets and therefore any styles would also need to be copied to the PiP window. We cannot use CSS media queries here.

Reply:
Thinking out loud, shall we rename this method to `copyToPictureInPictureWindow` then so that it's clear what it's doing?

Reply:
I think we should rename this because `requestPictureInPicture` on the video element has different parameters and a different underlying algorithm. It would make it clear to developers exactly what this is doing.

Reply:
I think we should resolve the naming issue outside of this PR, so I have clarified in the explainer that the name will likely change and added a tracking issue here: https://github.com/WICG/picture-in-picture/issues/129